presto-orc/src/main/java/com/facebook/presto/orc/writer/LongDictionaryColumnWriter.java

```diff
@@ -0,0 +1,300 @@
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.orc.OrcEncoding.DWRF;
+import static com.facebook.presto.orc.metadata.ColumnEncoding.ColumnEncodingKind.DICTIONARY;
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.DICTIONARY_DATA;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.util.Arrays.asList;
+
+public class LongDictionaryColumnWriter
+        extends DictionaryColumnWriter
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(LongDictionaryColumnWriter.class).instanceSize();
+    private static final int UNUSED_VALUE = -1;
+
+    private final LongOutputStream dictionaryDataStream;
+    private final int typeSize;
+
+    private LongDictionary dictionary;
+    private IntegerStatisticsBuilder statisticsBuilder;
+    private ColumnEncoding columnEncoding;
+    private LongColumnWriter directColumnWriter;
+    private long[] directValues;
+    private boolean[] directNulls;
+
+    public LongDictionaryColumnWriter(
+            int column,
+            Type type,
+            CompressionParameters compressionParameters,
+            Optional<DwrfDataEncryptor> dwrfEncryptor,
+            OrcEncoding orcEncoding,
+            MetadataWriter metadataWriter)
+    {
+        super(column, type, compressionParameters, dwrfEncryptor, orcEncoding, metadataWriter);
+        checkArgument(orcEncoding == DWRF, "Long dictionary encoding is only supported in DWRF");
+        checkArgument(type instanceof FixedWidthType, "Not a fixed width type");
+        this.dictionaryDataStream = new LongOutputStreamDwrf(compressionParameters, dwrfEncryptor, true, DICTIONARY_DATA);
+        this.dictionary = new LongDictionary(10_000);
+        this.typeSize = ((FixedWidthType) type).getFixedSize();
+        this.statisticsBuilder = new IntegerStatisticsBuilder();
+    }
+
+    @Override
+    protected int getInstanceSize()
+    {
+        return INSTANCE_SIZE;
+    }
+
+    @Override
+    public int getDictionaryEntries()
+    {
+        return dictionary.size();
+    }
+
+    @Override
+    public int getDictionaryBytes()
+    {
+        return dictionary.size() * typeSize;
+    }
+
+    @Override
+    protected ColumnWriter createDirectColumnWriter()
+    {
+        if (directColumnWriter == null) {
+            directColumnWriter = new LongColumnWriter(column, type, compressionParameters, dwrfEncryptor, orcEncoding, IntegerStatisticsBuilder::new, metadataWriter);
+        }
+        return directColumnWriter;
+    }
+
+    @Override
+    protected ColumnWriter getDirectColumnWriter()
+    {
+        return directColumnWriter;
+    }
+
+    @Override
+    protected boolean tryConvertToDirect(int rowGroupValueCount, IntBigArray rowGroupValues, int maxDirectBytes)
```

Comment:
Seems the tryConvertToDirect from slice dictionary writer can be used here as well. Any reason we avoid doing that?